tools/payloads/configbytes/apk_configbytes_extract.rb

```diff
@@ -0,0 +1,231 @@
+  check_tools(verbose)
+
+  temp_dir = Dir.mktmpdir
+  temp_dir << "/"
+
+  if options[:out].nil?
+    $stdout.puts "[-] No output option selected, working in #{temp_dir}"
+    output = "#{temp_dir}configbytes.txt"
+    options[:keep] = true
+  else
+    output = options[:out]
+  end
+
+  apk = options[:apk]
+  zip_file = "#{temp_dir}temp.zip"
+  temp_apk = "#{temp_dir}temp.apk"
+  FileUtils.cp(apk, temp_apk)
+
+  $stdout.puts "[+] Renaming apk file to zip file"
+  File.rename(temp_apk, zip_file)
+
+  $stdout.puts "[+] Using unzip on zip file for dex file"
+  unzip_cmd = run_cmd("unzip", "#{temp_dir}temp.zip -d #{temp_dir}", verbose)
+
+  $stdout.puts "[+] Using d2j-dex2jar on dex file to create jar file"
+  dex_file = "#{temp_dir}classes.dex"
+  d2j_cmd =  run_cmd("d2j-dex2jar", "-o #{temp_dir}classes.jar #{dex_file}", verbose)
+
+  $stdout.puts "[+] Using unzip on jar file for class files"
+  classes_dir = "#{temp_dir}classes/"
+  FileUtils.mkdir_p classes_dir
+  unzip_jar_cmd = run_cmd("unzip", "#{temp_dir}classes.jar -d #{classes_dir}", verbose)
+
+  $stdout.puts "[+] Using apktool to read AndroidManifest"
+  apktool_cmd = run_cmd("apktool", "d #{apk} -o #{temp_dir}decompile/", verbose)
+
+  amanifest = parse_manifest("#{temp_dir}decompile/AndroidManifest.xml")
+  package_path = amanifest.xpath("//manifest").first['package'].gsub(/\./, "/")
+  $stdout.puts "[+] Package path found: #{package_path}"
+
+  # DEFAULT METERPRETER PAYLOAD
+  payload_file = "#{classes_dir}#{package_path}/Payload.class"
+
+  # CREATING SEARCHABLE PAYLOAD
+  class_file = File.join("#{classes_dir}#{package_path}/?????", "?????.class")
+  class_files = Dir.glob(class_file)
+  searchable_payload = class_files[0]
+
+  # SETTING UP JAVA DECOMPILER
+  java_dir = "#{temp_dir}java/"
+  FileUtils.mkdir_p java_dir
+  fernflower_jar = options[:jar]
+
+  if File.exist?(payload_file) # SEARCH DEFAULT METASPLOIT APK
+    $stdout.puts "[+] Using fernflower to change class file to java file"
+    fernflower_cmd =  run_cmd("java", "-jar #{fernflower_jar} #{payload_file} #{java_dir}", verbose)
+
+    payload_file = File.dirname(payload_file)
+    $stdout.puts "[+] Class Path: #{payload_file}"
+
+    java_file ="#{java_dir}Payload.java"
+    $stdout.puts "[+] Metasploit Payload Class found!\n[+] #{java_file}"
+    extract_data(java_file, output)
+
+  elsif !searchable_payload.nil? # SEARCHING APK WTIH METERPRETER INJECTION
+    $stdout.puts "[+] Looking for Backdoored Metasploit Payload Classes"
+    $stdout.puts "[+] Using fernflower to change class files to java files"
+    for class_file in class_files
+      fernflower_cmd =  run_cmd("java", "-jar #{fernflower_jar} #{class_file} #{java_dir}", verbose)
+    end
+
+    class_path = File.dirname(class_file)
+    $stdout.puts "[+] Class Path: #{class_path}"
+
+    java_file = File.join("#{java_dir}", "?????.java")
+    java_files = Dir.glob(java_file)
+
+    for java_file in java_files
+      if File.read("#{java_file}").include? "byte"
```

Comment:
Any idea how prone to false positives this is? Have you tested it on other non-backdoored APKs?

Reply:
Yes, if an apk does not have an android meterpreter, a RuntimeError will be raised as it will not be able to find `default_payload_file` nor `searchable_payload`